flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/SourceQueryOperation.java

```diff
@@ -76,6 +78,25 @@ public String asSummaryString() {
                 "CatalogTable", args, getChildren(), Operation::asSummaryString);
     }
 
+    @Override
+    public String asSerializableString() {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(
+                String.format(
+                        "SELECT %s FROM %s",
+                        getResolvedSchema().getColumnNames().stream()
+                                .map(EncodingUtils::escapeIdentifier)
+                                .collect(Collectors.joining(", ")),
+                        getContextResolvedTable().getIdentifier().asSerializableString()));
+
+        if (dynamicOptions != null) {
+            stringBuilder.append(
+                    String.format("OPTIONS (%s)", OperationUtils.formatProperties(dynamicOptions)));
```

Comment:
this doesn't look like correct SQL. Did you mean SQL hints here?

Reply:
Yes, I did. However, after trying to fix it, I noticed there is actually no way to pass dynamic options from Table Api. I'll throw an exception for now if there are options set.